packages/compass-schema/src/stores/store.js

```diff
@@ -217,6 +222,62 @@ const configureStore = (options = {}) => {
       }
     },
 
+    _calculateDepthByPath(input, paths) {
+      if (!input) {
+        return paths;
+      }
+      input.forEach(({types, fields, path, bsonType}) => {
+        const increment = bsonType === 'Array' ? 1 : 0;
+        const score = path.split('.').length + increment;
+        if (!paths[path] || paths[path] < score) {
```

Comment:
This code seems a bit like it expects to visit the same `path` multiple times … does that actually happen?

Reply:
It does, given the structure of schema. In case of an array (`location.coordinates`), the path to an array item(`location.coordinates[0]`) is still the same because indexes are not counted and for that reason we have this increment of 1 for array. 
```
{
            name: 'location',
            path: 'location',
            types: [
              {
                bsonType: 'Document',
                path: 'location',
                fields: [
                  {
                    name: 'coordinates',
                    path: 'location.coordinates',
                    types: [
                      {
                        bsonType: 'Array',
                        path: 'location.coordinates',
                        types: [
                          {
                            bsonType: 'Double',
                            path: 'location.coordinates',
                          },
                        ],
                      }
                    ],
                    type: 'Array',
                  },
                  {
                    name: 'type',
                    path: 'location.type',
                    types: [
                      {
                        bsonType: 'String',
                        path: 'location.type',
                      }
                    ],
                    type: 'String',
                  },
                ],
              },
            ],
            type: 'Document',
          }
```

Reply:
It might be worth adding a comment about this fact :)